pkg/api/types.go

```diff
@@ -1662,6 +1662,83 @@ type SecretList struct {
 	Items []Secret `json:"items"`
 }
 
+// SecurityContext specifies the security constraints associated with a service account
+type SecurityContext struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Please split this into two objects - one that is what the pod uses (the security context the pod should run in) and one that is a declarative "this is the settings that are allowed".  I'll hold comments until you make that separation.